fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -209,17 +206,544 @@ fn expand_input_param(
+            "BurgundyBeefEnum".to_string(),
+            Property {
+                name: "unused".to_string(),
+                type_field: "enum SomeFrenchCuisine".to_string(),
+                components: None,
+            },
+        );
+        let abi_parser = ABIParser::new();
+        let result = expand_function(&the_function, &abi_parser, &custom_enums, &custom_structs);
+        // Some more editing was required because it is not rustfmt-compatible (adding/removing parentheses or commas)
+        let expected = TokenStream::from_str(
+            r#"
+#[doc = "Calls the contract's `HelloWorld` (0x00000000b6425163) function"]
+pub fn HelloWorld(
+    &self,
+    bim_bap_struct: BikeHelmet,
+    burgundy_beef_enum: SomeFrenchCuisine
+) -> ContractCall<((bool , u64 ,) , (bool, u64 ,))> {
+    Contract::method_hash(
+        &self.fuel_client,
+        &self.compiled,
+        [0, 0, 0, 0, 182, 66, 81, 99],
+        &[
+            ParamType::Struct(vec![ParamType::Bool, ParamType::U64]),
+            ParamType::Enum([Bool , U64])] , 
+            &[bim_bap_struct.into_token(), burgundy_beef_enum.into_token(),]
+    )
+    .expect("method not found (this should never happen)")
+}
+        "#,
+        );
+        let expected = expected.unwrap().to_string();
+        assert_eq!(result.unwrap().to_string(), expected);
+    }
+
+    // --- expand_selector ---
+    #[test]
+    fn test_expand_selector() {
+        let result = expand_selector(Selector::default());
+        assert_eq!(result.to_string(), "[0 , 0 , 0 , 0 , 0 , 0 , 0 , 0]");
+        let result = expand_selector(Selector::from([1, 2, 3, 4, 5, 6, 7, 8]));
+        assert_eq!(result.to_string(), "[1 , 2 , 3 , 4 , 5 , 6 , 7 , 8]");
+    }
+
+    // --- expand_fn_outputs ---
+    #[test]
+    fn test_expand_fn_outputs_zero_one_arg() {
+        let result = expand_fn_outputs(&[]);
+        assert_eq!(result.unwrap().to_string(), "()");
+        let result = expand_fn_outputs(&[Property {
+            name: "unused".to_string(),
+            type_field: "bool".to_string(),
+            components: None,
+        }]);
+        assert_eq!(result.unwrap().to_string(), "bool");
+        let result = expand_fn_outputs(&[Property {
+            name: "unused".to_string(),
+            type_field: String::from("struct streaming_services"),
+            components: Some(vec![
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("thistypedoesntexist"),
+                    components: None,
+                },
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("thistypedoesntexist"),
+                    components: None,
+                },
+            ]),
+        }]);
+
+        // the function has inconsistent  behavior for enum compared to struct:
+        // here we have to provide actual types in the components, not with the struct
```

Comment:
Yeah, I remember that. Do you mind creating an issue to eliminate this inconsistency? That's mostly because `expand_fn_outputs` doesn't explicitly handle enums, only structs:

```rust
if outputs[0].type_field.contains("struct ") {
    let tok: proc_macro2::TokenStream =
        extract_struct_name_from_abi_property(&outputs[0])
            .parse()
            .unwrap();
    Ok(tok)
} else {
    expand_type(&parse_param(&outputs[0])?)
}
```

Reply:
I had created issue #63

Reply:
Oh!! I didn't see it on the board because you forgot to add the Github Project to it 😅. No biggie though, just added to this one, just remember to do it on the next issues as well.